pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -156,31 +157,65 @@ func nodeToStartRevisionWith(getStaticPodState func(nodeName string) (state stat
 		}
 	}
 
-	// otherwise try to find a node that is not ready
+	// otherwise try to find a node that is not ready. Take the oldest one.
+	oldestNotReadyRevisionNode := -1
+	oldestNotReadyRevision := math.MaxInt32
 	for i := range nodes {
 		currNodeState := &nodes[i]
-		state, _, _, err := getStaticPodState(currNodeState.NodeName)
+		state, revision, _, err := getStaticPodState(currNodeState.NodeName)
 		if err != nil && apierrors.IsNotFound(err) {
 			return i, nil
 		}
 		if err != nil {
 			return 0, err
 		}
-		if state != staticPodStateReady {
+		revisionNum, err := strconv.Atoi(revision)
+		if err != nil {
 			return i, nil
 		}
+		if state != staticPodStateReady && revisionNum < oldestNotReadyRevision {
+			oldestNotReadyRevisionNode = i
+			oldestNotReadyRevision = revisionNum
+		}
+	}
+	if oldestNotReadyRevisionNode >= 0 {
+		return oldestNotReadyRevisionNode, nil
 	}
 
-	// last but not least, find a node that is has the wrong revision
+	// find a node that is has the wrong revision. Take the oldest one.
```

Comment:
what means "wrong" revision? you looking for a node that is on the oldest revision?